sdjournal/journal.go

```diff
@@ -291,6 +291,19 @@ func (j *Journal) Wait(timeout time.Duration) int {
 	return int(r)
 }
 
+// WaitIndefinitely will synchronously wait until the journal gets changed.  This
+// call will sleep indefinitely while it waits for a change.
+func (j *Journal) WaitIndefinitely() int {
+        j.mu.Lock()
+        // sd_journal_wait(3) calls for a (uint64_t) -1 to be passed to signify
+        // indefinite wait, but using a -1 overflows our C.uint64_t, so we use an
+        // equivalent hex value
+        r := C.sd_journal_wait(j.cjournal, (C.uint64_t)(0xffffffffffffffff))
```

Comment:
I'll let @vcaputo speak for himself but that's what I had in mind, except
with some sugar, e.g. could expose an `IndefiniteWait` sentinel const...

On Wed, Dec 23, 2015 at 6:44 PM, Christopher Snell <notifications@github.com
> wrote:

> In sdjournal/journal.go
> <https://github.com/coreos/go-systemd/pull/129#discussion_r48365192>:
>
> > @@ -291,6 +291,19 @@ func (j *Journal) Wait(timeout time.Duration) int {
> >  	return int(r)
> >  }
> >
> > +// WaitIndefinitely will synchronously wait until the journal gets changed.  This
> > +// call will sleep indefinitely while it waits for a change.
> > +func (j *Journal) WaitIndefinitely() int {
> > +        j.mu.Lock()
> > +        // sd_journal_wait(3) calls for a (uint64_t) -1 to be passed to signify
> > +        // indefinite wait, but using a -1 overflows our C.uint64_t, so we use an
> > +        // equivalent hex value
> > +        r := C.sd_journal_wait(j.cjournal, (C.uint64_t)(0xffffffffffffffff))
>
> I'm not following how you want this done. Wait() takes a time; do you want
> a WaitIndefinitely() helper function that passes 2540400h10m10.000000000s
> to Wait()?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/go-systemd/pull/129/files#r48365192>.
>